src/modules/collection/sagas.spec.ts

```diff
@@ -1844,11 +1845,42 @@ describe('when handling the fetch of collections', () => {
           [select(getWalletItems), []],
           [call([mockBuilder, 'fetchCollections'], undefined, mockedFetchParameters), [collection]]
         ])
-        .put(fetchCollectionsSuccess([collection], undefined))
+        .put(fetchCollectionsSuccess([collection], undefined, mockedFetchParameters))
         .dispatch(fetchCollectionsRequest(undefined, mockedFetchParameters))
         .run({ silenceTimeout: true })
     })
   })
+
+  describe('and should use the already fetched version', () => {
+    let mockedFetchParameters: FetchCollectionsParams
+    let alreadyFetchedCollections: Record<string, Collection>
+    beforeEach(() => {
+      collection = getCollectionMock()
+      mockedFetchParameters = {
+        assignee: '0x123',
+        isPublished: true,
+        q: collection.name,
+        sort: CurationSortOptions.NAME_ASC,
+        status: CurationStatus.TO_REVIEW
+      }
+      alreadyFetchedCollections = {
+        [collection.id]: collection
+      }
+    })
+
+    it('should put the success action with the data without pagination information', () => {
+      return expectSaga(collectionSaga, mockBuilder, mockBuilderClient, mockCatalyst)
+        .provide([
+          [select(getWalletItems), []],
+          [select(getLastFetchParams), mockedFetchParameters],
+          [select(getData), alreadyFetchedCollections],
+          [select(getCollectionPaginationData), undefined]
+        ])
+        .put(fetchCollectionsSuccess([collection], undefined, mockedFetchParameters))
+        .dispatch(fetchCollectionsRequest(undefined, mockedFetchParameters, true))
+        .run({ silenceTimeout: true })
+    })
```

Comment:
I think that it might be missing the case where the pagination data is required, which puts the success with pagination data, would you mind adding this case?

Reply:
added!